sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala

```diff
@@ -17,12 +17,21 @@
 
 package org.apache.spark.sql.catalyst
 
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths}
+
 import scala.util.control.NonFatal
 
 import org.apache.spark.sql.Column
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.test.SQLTestUtils
 
+/**
+ * A testsuite for LogicalPlan-to-SQL conversion.
```

Comment:
also add something like

"Each query has a golden generated SQL file in test/resources/sqlgen. The test suite also has built-in functionality to automatically generate these golden files. To do so, set the saveQuery flag to true and
run the test suite. After generating the golden files, make sure to set the flag to false."
```